src/repeater/index.tsx

```diff
@@ -1,23 +1,45 @@
-import React from 'react'
+import React from 'react';
 
-interface RepeaterProps<T> {
+interface RepeaterBasic<T> {
   items: T[];
   renderItem: (item: T, index: number) => JSX.Element;
   renderEmpty?: () => JSX.Element;
 }
 
-const Repeater = <T extends unknown>(props: RepeaterProps<T>) => {
+interface RepeaterMetaProps<T, K> {
+  items: { data: T[]; meta: K };
+  renderItem: (item: T & Partial<K>, index: number) => JSX.Element;
+  renderEmpty?: () => JSX.Element;
+}
+
+const Repeater = <T extends unknown, K extends Record<string, unknown>>(
+  props: RepeaterMetaProps<T, K> | RepeaterBasic<T>
+) => {
   const { items, renderItem, renderEmpty } = props;
 
-  if (items.length === 0) {
-    return renderEmpty ? renderEmpty() : null;
-  } else {
+  if ('data' in items && 'meta' in items) {
+    const { data, meta } = items;
+    if (data.length === 0) {
+      return renderEmpty ? renderEmpty() : null;
+    }
     return (
       <>
-        {items.map((item, index) => renderItem(item, index))}
+        {data.map((item, index) =>
+          typeof item === 'object' ? renderItem({ ...item, ...meta }, index) : renderItem(item as T & Partial<K>, index)
```

Comment:
We get the data as 
```js
{ data: [], meta: { pagination: {} } }
```

We merge individual item with it's meta. So, it can be accessed as 

```js
{
   "type": "expr",
    "content": "context_3iuex?.pagination?.total"
 }
```

The reason for taking the merging approach
- We don't need to maintain additional identifiers for the `meta`. Since we need `unique` name to make the closure access work.
- Second, if we separate them, Again it leads to changes in how the generators can return them in calback
```jsx
 <Repeater
  items={params}
  renderItem={(context_3iuex) => (
    <>
      <h3 className="home-text">
        {context_3iuex?.title?.rendered}
      </h3>
      <span className="home-text1">
        {context_3iuex?.description}
      </span>
      <div className="home-container1">
        <h4>Total</h4>
        <span>{context_3iuex?.pagination?.total}</span>
      </div>
    </>
  )}
/>

```

This will lead to 

```jsx
renderItem={(context_3iuex, context_3iuex_meta) => (
```

Makes it little difficult for our current requirement.